diff --git a/package.json b/package.json
index bc79549..ebbd0ec 100644
--- a/package.json
+++ b/package.json
@@ -9,6 +9,7 @@
     "axios": "^0.20.0",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
+    "react-router": "^5.2.0",
     "react-router-dom": "^5.2.0",
     "react-scripts": "3.4.3"
   },
diff --git a/public/index.html b/public/index.html
index df8eeb5..5f5467c 100644
--- a/public/index.html
+++ b/public/index.html
@@ -15,7 +15,6 @@
       integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T"
       crossorigin="anonymous"
     />
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
     <!--
       Notice the use of %PUBLIC_URL% in the tags above.
       It will be replaced with the URL of the `public` folder during the build.
@@ -24,7 +23,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React Pagination</title>
+    <title>Github Issues Viewer</title>
   </head>
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
diff --git a/src/App.js b/src/App.js
index 54332da..5ad21cc 100644
--- a/src/App.js
+++ b/src/App.js
@@ -1,5 +1,14 @@
 import React, { useState, useEffect } from 'react';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Link,
+  useParams
+} from "react-router-dom";
+
 import Issues from './components/Issues';
+import IssueDetail from './components/IssueDetails';
 import Pagination from './components/Pagination';
 import axios from 'axios';
 import './App.css';
@@ -31,17 +40,47 @@ const App = () => {
   const paginate = pageNumber => setCurrentPage(pageNumber);
 
   return (
-    <div className='container mt-5'>
-      <h1 className='text-primary mb-3'>Issue Browser</h1>
-      <Issues issues={currentIssues} loading={loading} />
-  
-      <Pagination
-        issuesPerPage={issuesPerPage}
-        totalIssues={issues.length}
-        paginate={paginate}
-      />
-    </div>
+    <Router>
+      <Switch>
+            <Route path="/issue/:id">
+              <IssueDetailView />
+            </Route>
+            <Route path="/">
+              <HomeView />
+            </Route>
+      </Switch>
+    </Router>
   );
+
+
+  function HomeView() {
+    return (
+      <div className='container mt-5'>
+        <h1 className='text-primary mb-3'>Issue Browser</h1>
+        <Issues issues={currentIssues} loading={loading} />
+    
+        <Pagination
+          issuesPerPage={issuesPerPage}
+          totalIssues={issues.length}
+          paginate={paginate}
+        />
+    </div>
+    );
+  }
+
+  function IssueDetailView() {
+    let { id } = useParams();
+    let current = null;
+    for(let issue in currentIssues) {
+      if (currentIssues[issue].number == id) {
+        current = currentIssues[issue];
+        break;
+      }
+    }
+    return (
+      <IssueDetail data={current} />
+    )
+  }
 };
 
 export default App;
diff --git a/src/components/IssueDetails.js b/src/components/IssueDetails.js
index 4a9eda1..9d751ee 100644
--- a/src/components/IssueDetails.js
+++ b/src/components/IssueDetails.js
@@ -1,18 +1,12 @@
-import React from 'react';
-
-const IssueDetail = ({ issues }) => {
-  if (loading) {
-    return <h2>Loading...</h2>;
-  }
+import React, {useState} from 'react';
 
+const IssueDetail = ({data}) => {
+  
   return (
-    <ul className='list-group mb-4'>
-      {issues.map(issue => (
-        <li key={issue.number} className='list-group-item'>
-            {issue}
-        </li>
-      ))}
-    </ul>
+    <div className='container mt-5'>
+        {JSON.stringify(data)}
+             
+    </div>
   );
 };
 
diff --git a/src/components/Issues.js b/src/components/Issues.js
index ce1a8bf..348e153 100644
--- a/src/components/Issues.js
+++ b/src/components/Issues.js
@@ -1,5 +1,9 @@
 import React from 'react';
 
+import {
+  Link
+} from "react-router-dom";
+
 const Issues = ({ issues, loading }) => {
   if (loading) {
     return <h2>Loading...</h2>;
@@ -8,14 +12,14 @@ const Issues = ({ issues, loading }) => {
   return (
     <ul className='list-group mb-4'>
       {issues.map(issue => (
+        <Link to={ "issue/" + issue.number  }>
         <li key={issue.number} className='list-group-item'>
             <a href='!#' className='page-link'>
-            
             {issue.number}. {issue.title}
             {issue.state}
-            </a>
-          
-        </li>
+            </a>        
+          </li>
+        </Link>
       ))}
     </ul>
   );
